pagopa/io-app
# chore: [IOBP-665,IOBP-570] PayPal metadata and latest used payment method adaption

## Short description
This PR adds the metadata for fast PayPal payment by adding the masked email and the pspBusinessName if both are present. In addition, the selector has been adapted to handle the “recently used” logic by taking into account the `lastUsage` attribute of the `clients` map

## List of changes proposed in this pull request
- Upgraded the new openapi for wallet & ecommerce
- Changed the selector to filter the "recently used" payment method following the new openapi definitions
- Added metadata to the "Pagamento veloce con PayPal" with the pspBusinessName and maskedEmail if both present.

## How to test
- Checkout this PR: https://github.com/pagopa/io-dev-api-server/pull/384
- Start a new payment and when you are in the payment selection step, you should be able to see a pre-selected method. If in the list of saved methods or the "recently used" there is a "Pagamento veloce con PayPal", you should be able even to see  metadata in the list item description


## Preview
<img src="https://github.com/pagopa/io-app/assets/34343582/f2dcd4ae-276d-4f32-b6bf-81fd75911ca3" width="350px" />
